src/main/java/core/basesyntax/dao/ma/CoachDaoImpl.java

```diff
@@ -2,7 +2,9 @@
 
 import core.basesyntax.model.ma.Coach;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
 
 public class CoachDaoImpl extends PersonDaoImpl implements CoachDao {
```

Comment:
how about refactoring `save()` method in PersonDaoImpl using transaction?

Reply:
done